src/decode.rs

```diff
@@ -122,6 +122,9 @@ pub struct Decoded<S> {
     /// buffer for the most-recently received line, pending completion (by a newline terminator) or
     /// extension (by more non-newline bytes)
     incomplete_line: Option<Vec<u8>>,
+    /// flagged if the last character processed as a carriage return; used to help process CRLF
```

Comment:
The first few commits are from the other PR. I based the test harness work off of it since it needed better validation.